AcademyResidentInformationApi.Tests/V1/Gateways/AcademyCouncilTaxGatewayTests.cs

```diff
@@ -85,35 +86,265 @@ public void GetCouncilTaxPayerInformationByAccountRefReturnsDetailsWithContactIn
+            response.EmailList.Should().BeEquivalentTo(testEmail);
             response.PhoneNumberList.Should().BeEquivalentTo(testPhone);
         }
 
-        private TaxPayer AddTaxPayerDatabaseRecord(int? accountRef, string firstname = null, string lastname = null)
+        [Test]
+        public void GetAllTaxPayersIfThereAreNoTaxPayersReturnsAnEmptyList()
+        {
+            _classUnderTest.GetAllTaxPayers().Should().BeEmpty();
+        }
+
+        [Test]
+        public void GetAllTaxPayersIfThereAreTaxPayersReturnsAListOfAccountNumbersAndNames()
+        {
+            var databaseEntity1 = AddTaxPayerDatabaseRecord();
+            var databaseEntity2 = AddTaxPayerDatabaseRecord();
+
+            AddPropertyInformationForTaxPayer(databaseEntity1.AccountRef);
+            AddPropertyInformationForTaxPayer(databaseEntity2.AccountRef);
+            AddContactInformationForTaxPayer(databaseEntity1.AccountRef);
+            AddContactInformationForTaxPayer(databaseEntity2.AccountRef);
+
+
+            var response = _classUnderTest.GetAllTaxPayers();
+
+            response.Should().BeOfType<List<TaxPayerInformation>>();
+            response.First().Should().BeEquivalentTo(databaseEntity1);
+            response.Last().Should().BeEquivalentTo(databaseEntity2);
+        }
+
+        [Test]
+        public void GetAllTaxPayersReturnsTaxPayersWithCouncilPropertyDetails()
+        {
+            var databaseEntity = AddTaxPayerDatabaseRecord();
+            var property = AddPropertyInformationForTaxPayer(databaseEntity.AccountRef);
+            AddContactInformationForTaxPayer(databaseEntity.AccountRef);
+
+
+            var response = _classUnderTest.GetAllTaxPayers();
+            response
+                .First().TaxPayerAddress.AddressLine1
+                .Should().BeEquivalentTo(property.AddressLine1);
+        }
+
+        [Test]
+        public void GetAllTaxPayersReturnTaxPayersWithContactDetails()
+        {
+
+            var testEmail = new List<string> { "test@email1.com", "test@email2.com" };
+            var testPhone = new List<string> { "00000000000", "122223333331", "22222222222", null };
+
+            var databaseEntity = AddTaxPayerDatabaseRecord();
+            AddPropertyInformationForTaxPayer(databaseEntity.AccountRef);
+            AddContactInformationForTaxPayer(databaseEntity.AccountRef, testEmail, testPhone);
+
+
+            var response = _classUnderTest.GetAllTaxPayers();
+            response.First().EmailList.Should().BeEquivalentTo(testEmail);
+            response.First().PhoneNumberList.Should().BeEquivalentTo(testPhone);
+        }
+
+        [Test]
+        public void GetAllTaxPayersWithFirstNameQueryParameterReturnsMatchingTaxPayer()
+        {
+            var databaseEntity1 = AddTaxPayerDatabaseRecord(firstname: "ciasom");
+            AddPropertyInformationForTaxPayer(databaseEntity1.AccountRef);
+            AddContactInformationForTaxPayer(databaseEntity1.AccountRef);
```

Comment:
These 3 lines are used a lot in the next few tests. Maybe there is some sort of refactoring that could be done?

Reply:
Good point, I've created a SetUpFullTaxPayerDatabaseRecord method and refactored the code.